### PR TITLE
A little bit of stuff I have done.

### PR DESCRIPTION
**Pull request by Makkkkus**
Ready for merge

### CODE REVIEW

**Comment by Makkkkus** [comment]
Oh, and this update fixes shearing and Spanish encoding. Plus it adds a cool play menu

File: Bugs and things to do.txt [hunk 1]
```diff
@@ -21,12 +21,6 @@ Note to self: for more little bugs/ideas/concerns, search code for "TODO"
 
 Make an Interactable interface, and both Entity AND Tile implement it, because both are interactable. Interactable holds methods for things like collision detection, tangibility, events for touching, and events for player key presses like interact ("v") and attack (that will also pickup furniture; it's sort of like, "swipe"). There will be a dedicated button to open the inventory. intereact will open chests, activate tnt, etc.
 
-****
-
-change "xa" and "ya" to something else; b/c they are NOT accelerations! They're distances! Specifically, they are the distance that a given entity is about to move.
```
Comment (chrisj42) [inline Bugs and things to do.txt:26]: this seems like a good point; why remove it?

File: ChangeLog.txt [hunk 2]
```diff
@@ -1,17 +1,19 @@
 Minicraft+ version 2.0.7
 
-* Farm tiles will no longer be destroyed if an item is dropped onto it
-* Made crops grow slower
-* Updated some textures; made wood, apples and gold apples look better
-* Made sheeps drop beef
 + Added sheep shearing
-+ Added shear
++ Added shearing
```
Comment (chrisj42) [inline ChangeLog.txt:4]: that's redundant, it would be better off as "added shear tool"

File: ChangeLog.txt [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 + Added baked potatoes as food
 + Made zombies drop potatoes
++ Added a play menu with Multiplayer and Singleplayer as options
 * Sparks (the airwizard bullets) have been optimised
 * Fixed picked up dungeon chests
 * Tiles now connect to the edge of the world
+* Farm tiles will no longer be destroyed if an item is dropped onto it
+* Made crops grow slower
+* Updated some textures; made wood, apples and gold apples look better
+* Made sheeps drop beef
```
Comment (chrisj42) [inline ChangeLog.txt:15]:
```suggestion
* Made sheep drop beef
```

File: README.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
-# minicraft-plus-revived
-Minicraft+, a modded version of minicraft that adds many more features to the original version. To learn more about minicraft, or play it yourself, take a look at [www.playminicraft.com](https://www.playminicraft.com).
+# Minicraft Plus
+Minicraft+ is a modded version of Minicraft that adds many more features to the original version. To learn more about Minicraft, or play it yourself, take a look at all the [releases](https://github.com/chrisj42/minicraft-plus-revived/releases).
```
Comment (chrisj42) [inline README.md:2]: I kinda feel bad not referencing the website, also, it's there because the site does in fact provide a lot of history on the game, not gone into here. Also, the download button links straight back to this releases page anyway.

File: README.md [hunk 5]
```diff
@@ -14,16 +14,45 @@ Here is a small share of the more major features that this mod adds to the origi
...
-* Expanded storyline: 4th dungeon level, and Second Boss
+* An expanded storyline: 4th dungeon level, and Second Boss
 * Bigger/Resizable screen size
 * Key Binding customization, so you can change the key controls
 * Tutorials
 * Texture Packs
 * And best of all, active development and community!
+* Localization: Different languages
+
+You can find a more comprehensive list of the earlier versions, before 1.9, in [this minecraft forum post](http://www.minecraftforum.net/forums/off-topic/general-gaming/452036-v1-6-minicraft-plus). Be aware however, that it's old; it's from many years ago before the project was abandoned.
```
Comment (chrisj42) [inline README.md:25]: you've added this, but haven't removed the other mention of it... why is this here?

File: README.md [hunk 6]
```diff
@@ -14,16 +14,45 @@ Here is a small share of the more major features that this mod adds to the origi
...
 
 You can find a more comprehensive list of the earlier versions, before 1.9, in [this minecraft forum post](http://www.minecraftforum.net/forums/off-topic/general-gaming/452036-v1-6-minicraft-plus). Be aware however, that it's old; it's from many years ago before the project was originally abandoned, and then subsequently rediscovered 5 years later. :3
 
-#### Goals
+#### Far, far away...
+
+These are simply **ideas**, and have no guarantee of being implemented any time soon, or even at all. Could be becuase they're complicated, or could be because I just haven't bothered considering them yet
+If their implementation becomes more likely, they will be moved to a different section.
 
-For goals and bugs, as well as ideas, read "Bugs and things to do.txt", and the stuff in the "ideas" folder.
```
Comment (chrisj42) [inline README.md:29]: I still think this is good to know.

File: src/minicraft/entity/Arrow.java [hunk 7]
```diff
@@ -43,19 +43,16 @@ public String getData() {
...
 		if (x < 0 || x>>4 > level.w || y < 0 || y>>4 > level.h) {
-			remove(); // remove when out of bounds
+			remove(); // Remove when out of bounds
 			return;
 		}
 
 		x += dir.getX() * speed;
 		y += dir.getY() * speed;
-		
-		// TODO I think I can just use the xr yr vars, and the normal system with touchedBy(entity) to detect collisions instead.
```
Comment (chrisj42) [inline src/minicraft/entity/Arrow.java:53]: as much as old TODOs are in fact old, do you know what this is referencing?

File: src/minicraft/entity/Entity.java [hunk 8]
```diff
@@ -36,7 +36,7 @@
 	public int x, y; // x, y entity coordinates on the map
 	private int xr, yr; // x, y radius of entity
 	private boolean removed; // Determines if the entity is removed from it's level; checked in Level.java
-	protected Level level; // the level that the entity is on
+	public Level level; // the level that the entity is on
```
Comment (chrisj42) [inline src/minicraft/entity/Entity.java:39]: I wholly disappove of increasing visibility for no reason, encapsulation is a more better coding practice.

File: src/minicraft/entity/Entity.java [hunk 9]
```diff
@@ -99,21 +99,25 @@ public Entity(int xr, int yr) { // add color to this later, in color update
...
 	public boolean interact(Player player, @Nullable Item item, Direction attackDir) {
-		if(item != null)
-			return item.interact(player, this, attackDir);
 		return false;
 	}
 	
 	/** Moves an entity horizontally and vertically. Returns whether entity was unimpeded in it's movement.  */
-	public boolean move(int xa, int ya) {
-		if(Updater.saving || (xa == 0 && ya == 0)) return true; // pretend that it kept moving
+	public boolean move(int xmov, int ymov) {
```
Comment (chrisj42) [inline src/minicraft/entity/Entity.java:115]: name them xd/yd instead, stands for delta/distance. It's more standard

File: Bugs and things to do.txt (hunk 1 above)
Comment (chrisj42) [reply Bugs and things to do.txt:26]: ah, you implemented it

File: src/minicraft/entity/furniture/Chest.java [hunk 10]
```diff
@@ -56,11 +58,12 @@ public void populateInvRandom(String lootTable, int depth) {
 			e.printStackTrace();
 		}
 	}
-	
+
 	@Override
-	public void take(Player player) {
+	public boolean interact(Player player, @Nullable Item item, Direction attackDir) {
```
Comment (chrisj42) [inline src/minicraft/entity/furniture/Chest.java:63]: good idea

File: src/minicraft/entity/furniture/Spawner.java [hunk 11]
```diff
@@ -176,7 +171,7 @@ public boolean interact(Player player, Item item, Direction attackDir) {
 			
 			return true;
 		}
-		
+
```
Comment (chrisj42) [inline src/minicraft/entity/furniture/Spawner.java:174]: if there's one thing I hate it's having the whitespace on blank lines removed... then again that's a personal preference, and since I don't work on this anymore and it's not exactly consistent throughout anyway I won't bother with it.

File: src/minicraft/entity/mob/Player.java [hunk 12]
```diff
@@ -437,21 +437,11 @@ public void tick() {
 					level.dropItem(x, y, drop);
 			}
 			
-			if ((activeItem == null || !activeItem.used_pending) && (input.getKey("attack").clicked || input.getKey("pickup").clicked) && stamina != 0 && onFallDelay <= 0) { // this only allows attacks or pickups when such action is possible.
```
Comment (chrisj42) [inline src/minicraft/entity/mob/Player.java:440]: why was this removed?

Comment (chrisj42) [reply src/minicraft/entity/mob/Player.java:440]: oh I see, it was put elsewhere... I suppose that makes sense. But keep in mind that not addressing the used-pending status may result in some item bugs. Make sure you know what you're doing.

File: src/minicraft/item/Inventory.java [hunk 13]
```diff
@@ -55,27 +61,26 @@ public void add(Item item, int num) {
 	 * @param item Item to be added.
 	 */
 	public void add(int slot, Item item) {
-		//if (Game.debug) System.out.println("adding item to an inventory: " + item);
 		if(item instanceof PowerGloveItem) {
 			System.out.println("WARNING: tried to add power glove to inventory. stack trace:");
 			Thread.dumpStack();
-			return; // do NOT add to inventory
```
Comment (chrisj42) [inline src/minicraft/item/Inventory.java:62]: idk I thought this comment was useful

File: src/minicraft/item/Item.java [hunk 14]
```diff
@@ -28,24 +28,13 @@ protected Item(String name, Sprite sprite) {
 		this.sprite = sprite;
 	}
 
-	/** Renders an item (sprite & name) in an inventory */
-	public void renderInventory(Screen screen, int x, int y, int fontColor) {
```
Comment (chrisj42) [inline src/minicraft/item/Item.java:32]: ...where did these methods go?

Comment (Makkkkus) [reply src/minicraft/item/Item.java:32]: Never used

File: src/minicraft/item/StackableItem.java [hunk 15]
```diff
@@ -39,7 +39,7 @@
 	}
 	
 	public int count;
-	// public int maxCount; // TODO I want to implement this later.
+	public int maxCount = 999; // TODO I want to implement this later.
```
Comment (chrisj42) [inline src/minicraft/item/StackableItem.java:42]: is this implemented, or did you give it a value but never use it? Also, why is it instance bound if it doesn't change?

File: src/minicraft/item/ToolItem.java [hunk 16]
```diff
@@ -52,7 +64,7 @@ public boolean isDepleted() {
 	
 	/** You can attack mobs with tools. */
 	public boolean canAttack() {
-		return true;
+		return type != ToolType.Shear;
```
Comment (chrisj42) [inline src/minicraft/item/ToolItem.java:67]: I mean... technically speaking, getting hit with a pair of shears would still hurt lol

Comment (Makkkkus) [reply src/minicraft/item/ToolItem.java:67]: So you don't kill the sheep

File: src/minicraft/level/tile/TreeTile.java [hunk 17]
```diff
@@ -102,8 +102,8 @@ public void hurt(Level level, int x, int y, int dmg) {
 
 		level.add(new TextParticle("" + dmg, x*16+8, y*16+8, Color.RED));
 		if (damage >= treeHealth) {
-			level.dropItem(x*16+8, y*16+8, 1, 2, Items.get("Wood"));
-			level.dropItem(x*16+8, y*16+8, 1, 2, Items.get("Acorn"));
+			level.dropItem(x*16+8, y*16+8, 1, 3, Items.get("Wood"));
+			level.dropItem(x*16+8, y*16+8, 0, 2, Items.get("Acorn"));
```
Comment (chrisj42) [inline src/minicraft/level/tile/TreeTile.java:106]: I suppose it makes sense to allow zero, since people probably end up with a vast quantity of acorns.

File: src/minicraft/saveload/Save.java [hunk 18]
```diff
@@ -297,7 +308,7 @@ public static String writeEntity(Entity e, boolean isLocalSave) {
 		
 		// don't even write ItemEntities or particle effects; Spark... will probably is saved, eventually; it presents an unfair cheat to remove the sparks by reloading the Game.
 		
-		//if(e instanceof Particle) return ""; // TODO I don't want to, but there are complications.
+		if(e instanceof Particle && Game.debug) System.out.println("Saving particle with id: " + e.eid);
```
Comment (chrisj42) [inline src/minicraft/saveload/Save.java:311]: might be worth looking into before you remove the comment.

File: src/minicraft/screen/TitleDisplay.java [hunk 19]
```diff
@@ -37,21 +37,12 @@ public TitleDisplay() {
...
-				else
-					Game.setMenu(new WorldGenDisplay());
-			}),
-			new SelectEntry("Join Online World", () -> Game.setMenu(new MultiplayerDisplay())),
+			new SelectEntry("Play", () -> Game.setMenu(new PlayDisplay())),
 			new SelectEntry("Options", () -> Game.setMenu(new OptionsDisplay())),
 			displayFactory("Help",
 				new SelectEntry("Instructions", () -> Game.setMenu(new BookDisplay(BookData.instructions))),
 				new BlankEntry(),
-				new SelectEntry("Storyline Guide (for the weak)", () -> Game.setMenu(new BookDisplay(BookData.storylineGuide))),
```
Comment (chrisj42) [inline src/minicraft/screen/TitleDisplay.java:54]: how dare u

lol

File: src/minicraft/screen/TitleDisplay.java [hunk 20]
```diff
@@ -242,13 +224,12 @@ public void render(Screen screen) {
 		"Milk is an idea!",
 		"Creeper, aw man",
```
Comment (chrisj42) [inline src/minicraft/screen/TitleDisplay.java:225]: ...I won't insert my personal biases here... but god is this cringe

File: src/minicraft/screen/TitleDisplay.java [hunk 21]
```diff
@@ -242,13 +224,12 @@ public void render(Screen screen) {
...
 		"So we back in the mine,",
-		"pickaxe swinging from side to side",
+		"Pickaxe swinging from side to side",
 		"In search of Gems!",
 		"Life itself suspended by a thread",
 		"saying ay-oh, that creeper's KO'd!",
 		"Gimmie a bucket!",
 		"Farming with water!",
-		"Press \"R\"!",
```
Comment (chrisj42) [inline src/minicraft/screen/TitleDisplay.java:251]: Hey this actually does something you know. Have you ever tried actually pressing r? I'd keep that one in.

File: src/resources/localization/french_fr-fr.mcpl [hunk 22]
```diff
@@ -840,7 +840,7 @@ Charger un Monde
 New World
 Creer un Monde
 
-Join Online World
```
Comment (chrisj42) [inline src/resources/localization/french_fr-fr.mcpl:843]: keep in mind this outdates all the translations.

**Review by chrisj42 — CHANGES_REQUESTED** [review]
Alright there's a lot of good content here, but there are also a number of things I would like you to look into. Hopefully you can see my comments in the "Files changed" menu, and can respond to each. Below are a couple general comments...

You realize that the docs folder exists for a reason? You can't just remove it. That's supporting an alternate website.

I like your usage of foreach loops, good fix.

File: src/minicraft/screen/TitleDisplay.java (hunk 21 above)
Comment (Makkkkus) [reply src/minicraft/screen/TitleDisplay.java:251]: Not anymore ;)

File: src/minicraft/item/StackableItem.java (hunk 15 above)
Comment (Makkkkus) [reply src/minicraft/item/StackableItem.java]: I was going to implement it; I guess I just forgot.

File: src/minicraft/screen/TitleDisplay.java (hunk 19 above)
Comment (Makkkkus) [reply src/minicraft/screen/TitleDisplay.java:54]: Sacrifices had to be made

File: src/minicraft/screen/TitleDisplay.java (hunk 20 above)
Comment (Makkkkus) [reply src/minicraft/screen/TitleDisplay.java]: Good point

File: src/resources/localization/french_fr-fr.mcpl (hunk 22 above)
Comment (Makkkkus) [reply src/resources/localization/french_fr-fr.mcpl:843]: Translations should not be much different

**Review by Makkkkus — COMMENTED** [review]
Done

File: src/minicraft/entity/mob/MobAi.java [hunk 23]
```diff
@@ -100,11 +100,11 @@ public void render(Screen screen) {
 	}
 	
 	@Override
-	public boolean move(int xa, int ya) {
+	public boolean move(int xmov, int yd) {
```
Comment (africalinguist) [inline src/minicraft/entity/mob/MobAi.java:103]: This inconsistent variable naming is supremely annoying me. Why is it yd and not ymov?

Comment (Makkkkus) [reply src/minicraft/entity/mob/MobAi.java:103]: I wonder how that happened

**Review by Makkkkus — COMMENTED** [review]
Fixed up the requested changes

File: src/minicraft/core/Updater.java [hunk 24]
```diff
@@ -152,7 +152,7 @@ else if (isValidServer())
 				gameTime++;
 			}
 			
-			if (!isValidServer() || menu != null && !hasMenu)
+			if (!isValidServer() || menu != null && !hasMenu) // What?
```
Comment (Makkkkus) [inline src/minicraft/core/Updater.java:155]: Effectively just !isValidServer()
